bin/all_sky_search/pycbc_sngls_statmap

```diff
@@ -153,29 +140,71 @@ bg_far, fg_far = significance.get_far(
     fg_time,
     **significance_dict[ifo])
 
-# Cumulative array of exclusive background triggers and the number
-# of exclusive background triggers louder than each foreground trigger
-bg_far_exc, fg_far_exc = significance.get_far(
-    back_stat_exc,
-    fore_stat,
-    bkg_exc_dec_facs,
-    fg_time_exc,
-    **significance_dict[ifo])
-
 fg_far = significance.apply_far_limit(fg_far, significance_dict, combo=ifo)
 bg_far = significance.apply_far_limit(bg_far, significance_dict, combo=ifo)
-fg_far_exc = significance.apply_far_limit(fg_far_exc, significance_dict, combo=ifo)
-bg_far_exc = significance.apply_far_limit(bg_far_exc, significance_dict, combo=ifo)
 
 bg_ifar = 1. / bg_far
 fg_ifar = 1. / fg_far
-bg_ifar_exc = 1. / bg_far_exc
-fg_ifar_exc = 1. / fg_far_exc
 
 f['background/ifar'] = conv.sec_to_year(bg_ifar)
-f['background_exc/ifar'] = conv.sec_to_year(bg_ifar_exc)
+
 f.attrs['background_time'] = fg_time
 f.attrs['foreground_time'] = fg_time
+
+# Find foreground triggers with IFAR > 1 per livetime and remove from
+# the exclusive background (not equal, as that could give an infinite
+# loop if using count_n_louder)
+
+# Need to make copies for use as exclusive triggers as these will have
+# items removed from them, and don't want to overwrite the original
+fg_time_exc = copy.deepcopy(fg_time)
+fg_ifar_exc = copy.deepcopy(fg_ifar)
+bg_ifar_exc = copy.deepcopy(bg_ifar)
+back_stat_exc = copy.deepcopy(back_stat)
+bkg_exc_dec_facs = copy.deepcopy(bkg_dec_facs)
+
+# Record indices into all_trigs for the exclusive background
+back_exc_locs = numpy.arange(len(all_trigs.stat))
+
+while any(bg_ifar_exc > fg_time):
```

Comment:
if this is being done recursively, the probability that we end up removing several random noise triggers is too high
Hence replace by an input option eg 'exclude-ifar-threshold' (in years) with a sensible default value

Reply:
Or stick with 1/livetime default and don't exclude recursively ..